extensions/extensions-helm/aissemble-configuration-store-chart/templates/webhook-cleanup-job.yaml

```diff
@@ -0,0 +1,46 @@
+{{ if .Values.webhook.enable }}
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ index .Values "aissemble-quarkus-chart" "app" "name" }}-webhook-cleanup
+  annotations:
+    "helm.sh/hook": pre-delete, pre-upgrade
+    "helm.sh/hook-delete-policy": hook-succeeded
+spec:
+  template:
+    metadata:
+      name: {{ index .Values "aissemble-quarkus-chart" "app" "name" }}-webhook-cleanup
+    spec:
+      serviceAccountName: {{ .Values.webhook.serviceAccount.name }}
+      restartPolicy: OnFailure
+      containers:
+      - name: clean-webhook
+        image: {{ index .Values "aissemble-quarkus-chart" "deployment" "image" "repo" }}{{ index .Values "aissemble-quarkus-chart" "deployment" "image" "name" }}:{{ index .Values "aissemble-quarkus-chart" "deployment" "image" "tag" | default .Chart.AppVersion }}
+        imagePullPolicy: {{ index .Values "aissemble-quarkus-chart" "deployment" "image" "imagePullPolicy"}}
+        command:
+        - "/bin/sh"
+        - "-c"
+        - "curl -ik \
+          -X DELETE \
+          -H \"Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)\" \
+          -H \"Accept: application/json\" \
+          -H \"Content-Type: application/json\" \
+          https://kubernetes.default.svc/apis/admissionregistration.k8s.io/v1/mutatingwebhookconfigurations/{{ .Values.webhook.name }} \
+        {{- if .Values.webhook.certSecret.create }}
+          && \
+          curl -ik \
+          -X DELETE \
+          -H \"Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)\" \
+          -H \"Accept: application/json\" \
+          -H \"Content-Type: application/json\" \
+          https://kubernetes.default.svc/api/v1/namespaces/{{ .Release.Namespace }}/secrets/{{ .Values.webhook.certSecret.name }}
+        {{- end }}
+          && \
+          curl -ik \
+          -X DELETE \
+          -H \"Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)\" \
+          -H \"Accept: application/json\" \
+          -H \"Content-Type: application/json\" \
+          --data \"{\\\"kind\\\":\\\"DeleteOptions\\\",\\\"apiVersion\\\":\\\"batch/v1\\\",\\\"propagationPolicy\\\":\\\"Foreground\\\"}\" \
+          https://kubernetes.default.svc/apis/batch/v1/namespaces/{{ .Release.Namespace }}/jobs/{{ index .Values "aissemble-quarkus-chart" "app" "name" }}-webhook-init"
```

Comment:
Q: Why only delete the init job at delete/upgrade? Couldn't we delete it right after it runs?

Reply:
Update the hook to delete the init job on success. Left this curl delete here though so it still deletes the job on helm uninstall if the init job failed.